### PR TITLE
Test `load-data` ns

### PR DESCRIPTION
**Pull request by Eleonore9**
- Add tests for data loading & schema generation in `load-data` namespace
- Spaces, /, apostrophes in input data column names get replaced with '-'
- Add description & circle CI badge to README

### CODE REVIEW

File: test/witan/models/load_data_test.clj [hunk 1]
```diff
@@ -0,0 +1,67 @@
+(ns witan.models.load-data-test
+  (:require [witan.models.load-data :refer :all]
+            [clojure.test :refer :all]
+            [schema.core :as s]))
+
+(defn- same-coll? [coll1 coll2]
+  (= (set coll1) (set coll2)))
```
Comment (otfrom) [inline test/witan/models/load_data_test.clj]: are you sure about using set here?

```
user> (= (set [1 1 2]) (set [1 2]))
true
user> (= [1 1 2] [1 2])
false
```

File: test/witan/models/load_data_test.clj [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+(deftest custom-keyword-test
+  (testing "String column names are keywordised, with '.' and ' ' replaced by '-'."
+    (is (= [:col-1 :col2 :col-3]
+           (#'witan.models.load-data/custom-keyword ["col 1" "col2" "col.3"])))))
```
Comment (otfrom) [inline test/witan/models/load_data_test.clj]: testing private functions always makes me itchy. If they are private it is b/c they are part of the implementation and might change w/o changing the interface. I'd expect to be able to change a private function without having to change a test. Same with the use of `load-csv` below.

Comment (Eleonore9) [inline test/witan/models/load_data_test.clj]: Np, these functions don't need to be private.

File: test/witan/models/load_data_test.clj (hunk 1 above)
Comment (Eleonore9) [inline test/witan/models/load_data_test.clj]: I think we need to return as equal two collections with the same content but possibly of different type (vector or list) and ordered differently.

Comment (otfrom) [inline test/witan/models/load_data_test.clj]:
```
user> (= '(1 1 2) [1 1 2])
true
```

and won't order be significant? 

if you do want it w/o significant order then I'd do:

```
user> (= (sort '(1 1 2)) (sort [1 2 1]))
true
```

but converting to sets gets rid of duplicate values, which may or may not be what you want.

File: test/witan/models/load_data_test.clj [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  [m]
+  (some true? (map nil? (vals m))))
+
+(deftest load-dataset-test
+  (let [data-map (load-dataset :births-data test-file-1)
+        data (:births-data data-map)]
+    (testing "Loaded dataset has correct column names"
+      (is (= [:gss-code :sex :age :births :year] (ds/column-names data))))
+    (testing "Dataset was loaded"
+      (is (= nil (nils-in-map? data-map))))
```
Comment (otfrom) [inline test/witan/models/load_data_test.clj]: `(is (every? identity (vals m)))` instead? I _think_ you are saying there shouldn't be any nil values in the map.

File: test/witan/models/load_data_test.clj [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    (is (= ColumnsSchema (make-ordered-ds-schema [[:age s/Int] [:sex s/Str] [:year s/Int]
+                                                  [:gss-code s/Str] [:births s/Num]]))))) 
+(deftest make-row-schema-test
+  (testing "Generate row shema from column schema"
+    (is (= RowSchema (make-row-schema ColumnsSchema)))))
+
+(defn nils-in-map?
+  "Returns true if any values are nil, otherwise returns nil"
+  [m]
+  (some true? (map nil? (vals m))))
```
Comment (otfrom) [inline test/witan/models/load_data_test.clj]: this function probably isn't needed given `every?` below.

File: test/witan/models/load_data_test.clj (hunk 3 above)
Comment (LoraArmstrong) [inline test/witan/models/load_data_test.clj]: thanks. yes, that was the goal.

**Comment by otfrom** [comment]
:shipit:

**Pull request merged**